orca-clouddriver/src/main/groovy/com/netflix/spinnaker/orca/clouddriver/tasks/providers/gce/GoogleImageTagger.java

```diff
@@ -0,0 +1,192 @@
+/*
+ * Copyright 2017 Google, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.netflix.spinnaker.orca.clouddriver.tasks.providers.gce;
+
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.google.common.collect.ImmutableMap;
+import com.netflix.spinnaker.orca.clouddriver.OortService;
+import com.netflix.spinnaker.orca.clouddriver.tasks.image.ImageTagger;
+import com.netflix.spinnaker.orca.clouddriver.utils.CloudProviderAware;
+import com.netflix.spinnaker.orca.pipeline.model.Stage;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Component;
+
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.Collection;
+import java.util.HashMap;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+import java.util.concurrent.atomic.AtomicBoolean;
+import java.util.stream.Collectors;
+
+import static java.lang.String.format;
+
+@Component
+public class GoogleImageTagger implements ImageTagger, CloudProviderAware {
+  private static final Logger log = LoggerFactory.getLogger(GoogleImageTagger.class);
+  private static final Set<String> BUILT_IN_TAGS = new HashSet<>(Arrays.asList("appversion", "build_host"));
+
+  @Autowired
+  OortService oortService;
+
+  @Autowired
+  ObjectMapper objectMapper;
+
+  @Override
+  public ImageTagger.OperationContext getOperationContext(Stage stage) {
+    StageData stageData = (StageData) stage.mapTo(StageData.class);
+
+    Collection<MatchedImage> matchedImages = findImages(stageData.imageNames, stage);
+
+    stageData.imageNames = matchedImages.stream()
+      .map(matchedImage -> matchedImage.imageName)
+      .collect(Collectors.toSet());
+
+    // Built-in tags are not updatable
+    Map<String, String> tags = stageData.tags.entrySet().stream()
+      .filter(entry -> !BUILT_IN_TAGS.contains(entry.getKey()))
```

Comment:
Why is this ignored here? Shouldn't this be caught by the `Validator` in clouddriver?